framework/src/base/NonlinearSystemBase.C

```diff
@@ -278,6 +278,9 @@ NonlinearSystemBase::addKernel(const std::string & kernel_name, const std::strin
       _time_kernels.addObject(kernel, tid);
     else
       _non_time_kernels.addObject(kernel, tid);
+
+    // store kernels to warehouse, if necessary
```

Comment:
This comment doesn't seem right. It looks like we are always adding kernels in the `addExtraWarehouse()` routine. We have a lot of different warehouses for kernels now. It looks like at least 5 different warehouses. I wonder if we should consolidate this a bit and just add more methods to the warehouse for proper retrieval.